NAS2D/Renderer/Window.h

```diff
@@ -0,0 +1,91 @@
+#pragma once
+
+#include <string>
+#include <utility>
+
+enum class WindowDisplayMode {
+	Windowed
+	,Borderless
+	,Borderless_Fullscreen
+};
+
+struct WindowDesc {
+	//The client dimensions
+	//Window creation process should automatically add space for borders, menus and title bars
+	std::pair<int, int> dimensions{1600,600};
+	std::pair<int, int> position{0,0};
+	WindowDisplayMode mode{WindowDisplayMode::Windowed};
+};
```

Comment:
It would be more consistent to use `Point` and `Vector` here. The namespace prefix can be omitted if the header contents is wrapped in `namespace NAS2D`. Also, spaces after commas in data lists.
```cpp
struct WindowDesc
{
	Vector<int> dimensions{1600, 900};
	Point<int> position{0, 0};
	WindowDisplayMode mode{WindowDisplayMode::Windowed};
};
```

Might also make sense to move the comment to a function where that's more relevant. Data structures tend to be pretty dumb about details like that. It might be a class constructor or size method that would handle any sort of size adjustment, or documentation for a user about how they should specify parameters.